src/lite-yt-embed.js

```diff
@@ -54,6 +58,18 @@ class LiteYTEmbed extends HTMLElement {
         this.addEventListener('click', this.addIframe);
     }
 
+    static support_format_webp() {
```

Comment:
This canvas check is cool cuz its synchronous, however fails to return `true` for safari, which does support webp images.
(Its just their canvas toDataURL doesn't do webp exports. shrug.)

https://github.com/Modernizr/Modernizr/blob/master/feature-detects/img/webp.js has the other approach i'd normally gravitate towards: testing a datauri and looking for onload/onerror. (and apparently a 1px width check too!) downside is it's async, but perhaps that's OK to pick up safari support?

dunno if there's another reliable synchronous feature detect. i wouldn't expect one.

and of course, UA sniffing is an option but... nobody here _wants_ to do it. :)

Reply:
> the browser will automatically select the WebP version

in addition to picture there's -webkit-image-set which has similar type selection, but for background images.  (i have a slight preference for the background-image approach because it makes the progressive-enhancement markup easier... but tbh i might be too stubborn here :)

but anyway.. for some videos the webp doesnt exist. (see my below comment) and i thinkkkk neither of these two features will choose a diff image because a 404 came back. so.. i dont love that we have to use JS, but it seems like we do. (but i'd love to be proven wrong here)

Reply:
I wasn't expecting WebP version not to exist, do you have an example of that? SD WebP image I mean.